docs/tutorials/development.md

```diff
@@ -0,0 +1,151 @@
+`Close()`, which will be called on filter shutdown.
+
+The simplest filter possible is, if `filters.Spec` and
+`filters.Filter` are the same type:
+
+```
+type myFilter struct{}
+
+func NewMyFilter() filters.Spec {
+	return &myFilter{}
+}
+
+func (spec *myFilter) Name() string { return "myFilter" }
+
+func (spec *myFilter) CreateFilter(config []interface{}) (filters.Filter, error) {
+     return NewMyFilter(), nil
+}
+
+func (f *myFilter) Request(ctx filters.FilterContext) {
+     // change data in ctx.Request() for example
+}
+
+func (f *myFilter) Response(ctx filters.FilterContext) {
+     // change data in ctx.Response() for example
+}
+```
+
+Find a detailed example at [how to develop a filter](/skipper/reference/development#how-to-develop-a-filter).
+
+## Predicates
+
+Predicates allow to match a condition, that can be based on arbitrary
+HTTP data in the Request. There are also predicates, that use a chance
+`Traffic()` or the current local time, for example `After()`, to match
+a request and do not use the HTTP data at all.
+
+A predicate consists of at least two types `routing.Predicate`
+and `routing.PredicateSpec`, which are both interfaces.
+
+A spec will be created in the bootstrap procedure of a skipper
+process. A spec has to satisfy the `routing.PredicateSpec` interface
+`Name() string` and `Create([]interface{}) (routing.Predicate, error)`.
+
+The actual predicate implementation has to satisfy the
+`routing.Predicate` interface `Match(*http.Request) bool` and returns
+true if the predicate matches the request. If false is returned, the
+routing table will be searched for another route that might match the
+given request.
+
+The simplest possible predicate implementation is, if `routing.PredicateSpec` and
+`routing.Predicate` are the same type:
+
+```
+type myPredicate struct{}
+
+func NewMyPredicate() routing.PredicateSpec {
+	return &myPredicate{}
+}
+
+func (spec *myPredicate) Name() string { return "myPredicate" }
+
+func (spec *myPredicate) Create(config []interface{}) (routing.Predicate, error) {
+     return NewMyPredicate(), nil
+}
+
+func (f *myPredicate) Match(r *http.Request) bool {
+     // match data in *http.Request for example
+     return true
+}
+```
+
+Predicates are quite similar to implement as Filters, so for a more
+complete example, find an example [how to develop a filter](/skipper/reference/development#how-to-develop-a-filter).
+
+## Dataclients
+
+Dataclients are the way how to integrate new route
+sources. Dataclients pull information from a source and create routes
+for skipper's routing table.
+
+You have to implement `routing.DataClient`, which is an interface that defines
+function signatures `LoadAll() ([]*eskip.Route, error)` and
+`LoadUpdate() ([]*eskip.Route, []string, error)`.
+
+A complete example is the [routestring implementation](https://github.com/zalando/skipper/blob/master/dataclients/routestring/string.go), which fits in
+less than 50 lines of code.
```

Comment:
Do you it can be relevant how both polling and blocking are supported? Sg. like: 

The LoadUpdate() method can be implemented either in a way that returns immediately, or blocks until there is a change. The routing package will regularly call the LoadUpdate() method with a small delay between the calls.